### DISCUSSION TITLE
Access comments for a specific node

### DISCUSSION THREAD

**Discussion by arthurdenner** [babel/babel#14106] [Q&A] [answered]
Hi everyone!
I don't have deep knowledge of ASTs and decided to ask for help here.

I'm parsing a file and getting its exports with the following approach:

```js
/**
 * @deprecated use theNewMethod instead
 */
const oldMethod = () => {}

const theNewMethod = () => {}

export { oldMethod, theNewMethod }

/* ... */

const Parser = require('@babel/parser');
const traverse = require('@babel/traverse').default;
const fileContent = '...\nexport { oldMethod, theNewMethod }';
const ast = Parser.parse(fileContent, { sourceType: 'module' });

traverse(ast, {
  enter(path) {
    if (path.isExportSpecifier()) {
      const exportName = path.node.exported.name;
      // `exportName` is either 'oldMethod' or 'theNewMethod'

      // Is it possible to get the comments from this node?
    }
  },
});
```
Here is an AST Explorer link of my code above: [link](https://astexplorer.net/#/gist/bba86533f3239d040f676da0ea9721e0/8f67c70028ca7d9be04d3aec9d9d6c79c455adaf).

The goal with this was to find that "module X exports A, B and C."
Using `enter` + `path.isExportIdentifier` works fine, it's a simple solution.
However, if a method is deprecated, I'm annotating it with a JSDoc comment (`@deprecated`).

Is it possible to access the `leadingComments` for the `ExportSpecifier` or should I use another strategy?

Please let me know if the above isn't clear so I can try to elaborate.

Tips are very welcome! 😄

**Comment by JLHwung** [comment, marked as answer]
In your case, you can access the `@deprecated` comment from the leading comments of the _binding path_ of the name of the exported identifier.

```js
// let's say `path` is an ExportSpecifier
const bindingPath = path.scope.getOwnBinding(path.node.exported.name).path;
const isDeprecated = bindingPath.parent.leadingComments.some(comment => comment.value.includes("@deprecated"));
```

**Reply by arthurdenner** [reply]
Thank you very much for your help!

I tried your code and `bindingPath.node` was a `VariableDeclarator` node without `leadingComments`.
By changing it to `bindingPath.parent`, which is a `VariableDeclaration` node with `leadingComments`, it worked!
I can notice the same thing on the AST Explorer, `VariableDeclaration` is the one with `leadingComments`.

If that's expected and you don't mind, please edit your comment so I can mark it as the answer 🙌🏽
